app/src/components/market/create.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import { Textfield, Categories, Button } from '../common'
+import { useCreateMarket } from './hooks/use_create'
+
+const Div = styled.div`
+  height: 50px;
+  display: flex;
+  align-items: center;
+`
+
+export const CreateMarket: React.FC = () => {
+  const { values, handleChange, handleSubmit } = useCreateMarket({
```

Comment:
Why is this behind a custom hook? What are the odds we'll need to reuse this logic?

Reply:
The hook is in another file to keep the main file with fewer lines, and is generic because the form still does not do something specific. I think this will probably change in the next weeks because we will add more functionality...

Reply:
I think adding indirection just to reduce the average number of lines per file is not a good practice. The custom hook is not reusable nor testable, and it doesn't separate concerns. The only advantage of this over inlining everything is that a file is split in two (which is different from separating logic in two modules).

That being said, I don't think this is enough to block this PR.